src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -1222,4 +1223,1592 @@ struct S3
+    public required int Field1;
+    public required int Field2;
+}
+";
+
+        var creation = $@"C c = new{constructor}() {{ Prop2 = 1, Field2 = 1 }};";
+        var comp = CreateCompilationWithRequiredMembers(new[] { c, creation });
+
+        var expectedDiagnostics = constructor == " C" ? new[]
+        {
+            // (1,11): error CS9506: Required member 'C.Prop1' must be set in the object initializer or attribute constructor.
+            // C c = new C() { Prop2 = 1, Field2 = 1 };
+            Diagnostic(ErrorCode.ERR_RequiredMemberMustBeSet, "C").WithArguments("C.Prop1").WithLocation(1, 11),
+            // (1,11): error CS9506: Required member 'C.Field1' must be set in the object initializer or attribute constructor.
+            // C c = new C() { Prop2 = 1, Field2 = 1 };
+            Diagnostic(ErrorCode.ERR_RequiredMemberMustBeSet, "C").WithArguments("C.Field1").WithLocation(1, 11)
+        }
+        : new[] {
+            // (1,7): error CS9506: Required member 'C.Prop1' must be set in the object initializer or attribute constructor.
+            // C c = new() { Prop2 = 1, Field2 = 1 };
+            Diagnostic(ErrorCode.ERR_RequiredMemberMustBeSet, "new").WithArguments("C.Prop1").WithLocation(1, 7),
+            // (1,7): error CS9506: Required member 'C.Field1' must be set in the object initializer or attribute constructor.
+            // C c = new() { Prop2 = 1, Field2 = 1 };
+            Diagnostic(ErrorCode.ERR_RequiredMemberMustBeSet, "new").WithArguments("C.Field1").WithLocation(1, 7)
+        };
+
+        comp.VerifyDiagnostics(expectedDiagnostics);
+
+        var cComp = CreateCompilationWithRequiredMembers(c);
+        comp = CreateCompilation(creation, references: new[] { useMetadataReference ? cComp.ToMetadataReference() : cComp.EmitToImageReference() });
+
+        comp.VerifyDiagnostics(expectedDiagnostics);
+    }
+
+    [Theory]
+    [CombinatorialData]
+    public void EnforcedRequiredMembers_NoInheritance_AllSet(bool useMetadataReference, [CombinatorialValues("", " C")] string constructor)
+    {
+        var c = @"
+public class C
+{
+    public required int Prop1 { get; set; }
+    public required int Field1;
+}
+";
+
+        var creation = @"
+C c = new" + constructor + @"() { Prop1 = 1, Field1 = 1 };
+System.Console.WriteLine($""{c.Prop1}, {c.Field1}"");
+";
+        var comp = CreateCompilationWithRequiredMembers(new[] { c, creation });
+        CompileAndVerify(comp, expectedOutput: "1, 1");
+
+        var cComp = CreateCompilationWithRequiredMembers(c);
+        comp = CreateCompilation(creation, references: new[] { useMetadataReference ? cComp.ToMetadataReference() : cComp.EmitToImageReference() });
+        CompileAndVerify(comp, expectedOutput: "1, 1");
+    }
+
+    [Fact]
+    public void EnforcedRequiredMembers_NoInheritance_Unsettable()
+    {
+        var c = @"
+var c = new C();
+c = new C() { Prop1 = 1, Field1 = 1 };
+
+public class C
+{
+    public required int Prop1 { get; }
```

Comment:
Do we have a test with required unsettable property from metadata?

Reply:
Gentle ping

Reply:
I added one, just below this one.